config/stp.py

```diff
@@ -407,27 +637,68 @@ def stp_global_hello_interval(_db, hello_interval):
     db = _db.cfgdb
     check_if_global_stp_enabled(db, ctx)
     is_valid_hello_interval(ctx, hello_interval)
-    is_valid_stp_global_parameters(ctx, db, "hello_time", hello_interval)
-    update_stp_vlan_parameter(ctx, db, "hello_time", hello_interval)
-    db.mod_entry('STP', "GLOBAL", {'hello_time': hello_interval})
+    current_mode = get_global_stp_mode(db)
+
+    if current_mode == "pvst":
+        is_valid_stp_global_parameters(ctx, db, "hello_time", hello_interval)
+        update_stp_vlan_parameter(ctx, db, "hello_time", hello_interval)
+        db.mod_entry('STP', "GLOBAL", {'hello_time': hello_interval})
+    elif current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'hello_time': hello_interval})
+        update_mst_instance_parameters(ctx, db, 'hello_time', hello_interval)
+    else:
+        ctx.fail("Invalid STP mode configured")
 
 
 # cmd: STP global max age
+# MST CONFIGURATION IN THE STP_MST GLOBAL TABLE
+# config spanning_tree max_age <6-40 seconds>
 @spanning_tree.command('max_age')
 @click.argument('max_age', metavar='<6-40 seconds>', required=True, type=int)
 @clicommon.pass_db
 def stp_global_max_age(_db, max_age):
     """Configure STP global max_age"""
     ctx = click.get_current_context()
     db = _db.cfgdb
+    current_mode = get_global_stp_mode(db)
+    if current_mode == "pvst":        
+        check_if_global_stp_enabled(db, ctx)
+        is_valid_max_age(ctx, max_age)
+        is_valid_stp_global_parameters(ctx, db, "max_age", max_age)
+        update_stp_vlan_parameter(ctx, db, "max_age", max_age)
+        db.mod_entry('STP', "GLOBAL", {'max_age': max_age})
+    elif current_mode == "mst":
+        db.mod_entry('STP_MST', "GLOBAL", {'max_age': max_age})
+        update_mst_instance_parameters(ctx, db, 'max_age', max_age)
```

Comment:
We dnt need this instance update call

Reply:
I have addressed this comment, please review